src/ucp/core/ucp_ep.c

```diff
@@ -72,7 +73,24 @@ ucs_status_t ucp_ep_new(ucp_worker_h worker, uint64_t dest_uuid,
     ep->cfg_index        = ucp_worker_get_ep_config(worker, &key);
     ep->am_lane          = UCP_NULL_LANE;
     ep->flags            = 0;
-    ucs_queue_head_init(&ep->stream_data);
+
+    if (worker->context->config.features & UCP_FEATURE_STREAM) {
+        ep->ext.stream = ucs_calloc(1, sizeof(*ep->ext.stream),
+                                    "ucp ep stream extension");
+        if (ep->ext.stream == NULL) {
+            ucs_error("Failed to allocate ucp ep stream extension");
+            status = UCS_ERR_NO_MEMORY;
+            goto err;
```

Comment:
need to be 'err_free_ep' label. Also one more exit label which would clear this extension is needed